src/libcore/slice/mod.rs

```diff
@@ -400,14 +400,14 @@ impl<T> SliceExt for [T] {
         while size > 1 {
             let half = size / 2;
             let mid = base + half;
-            // mid is always in [0, size).
+            // mid is always in [0, size].
             // mid >= 0: by definition
             // mid < size: mid = size / 2 + size / 4 + size / 8 ...
             let cmp = f(unsafe { s.get_unchecked(mid) });
             base = if cmp == Greater { base } else { mid };
             size -= half;
         }
-        // base is always in [0, size) because base <= mid.
+        // base is always in [0, size] because base <= mid.
```

Comment:
... and here maybe spell out `base < size`?

Reply:
What do you mean by that?

Reply:
Was just thinking about how to make this line more explicit, too. But if you don't see the need to change it, that's fine by me too.